cmd/cluster/core/create.go

```diff
@@ -943,3 +952,43 @@ func parseTolerationString(str string) (*corev1.Toleration, error) {
 
 	return &toleration, nil
 }
+
+// validateMgmtClusterAndNodePoolCPUArchitectures checks if a multi-arch release image or release stream was provided.
+// If none were provided, checks to make sure the NodePool CPU arch and the management cluster CPU arch match; if they
+// do not, the CLI will return an error since the NodePool will fail to complete during runtime.
+func validateMgmtClusterAndNodePoolCPUArchitectures(ctx context.Context, opts *RawCreateOptions) error {
+	validMultiArchImage := false
+
+	// Check if the release image is multi-arch
+	if len(opts.ReleaseImage) > 0 && len(opts.PullSecretFile) > 0 {
+		pullSecret, err := os.ReadFile(opts.PullSecretFile)
+		if err != nil {
+			return fmt.Errorf("failed to read pull secret file: %w", err)
+		}
+
+		validMultiArchImage, err = registryclient.IsMultiArchManifestList(ctx, opts.ReleaseImage, pullSecret)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If not release image was provided, check if a release stream was provided instead and its multi-arch
+	if opts.ReleaseImage == "" && len(opts.ReleaseStream) > 0 && strings.Contains(opts.ReleaseStream, "multi") {
+		validMultiArchImage = true
+	}
+
+	// If a release image/stream is not multi-arch, check the mgmt & NodePool CPU architectures match
+	if !validMultiArchImage {
+		mgmtClusterCPUArch, err := hyperutil.GetMgmtClusterCPUArch(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to check mgmt cluster CPU arch: %v", err)
+		}
+
+		err = hyperutil.DoesMgmtClusterAndNodePoolCPUArchMatch(mgmtClusterCPUArch, opts.Arch)
```

Comment:
I think I would rather relax this req for now and not fail if opts.Arch is empty. This input is only leveraged to lookup an ami, which we could also infer by figuring the arch from the release image itself

Reply:
It's not only used for AMI for AWS. We also check the Arch is valid for a platform in `isArchAndPlatformSupported` in the NodePool Controller.

This field should never be empty since it gets a k8s default in the API and in the CLI.

We can't infer the arch from the release image when we are using multi-arch release images.

Reply:
I'm aware, I also think we should eventually drop this isArchAndPlatformSupported in favour of API validation and let that field have empty input. But anyways lets leave it as it's for now.

Reply:
Ack